wordpress-mobile/WordPress-Android
# [Notifications Refresh] 🤖 Milestone Details: hides the title for achievements

This PR covers just item 1 of https://github.com/Automattic/wordpress-mobile/issues/23 and introduces a new mechanism for identifying achievement/milestone notifications (ref p1714062709630809/1714052620.609979-slack-C06BWNSR02K).
The rest of the items will be covered in follow up PRs after refactoring part of the code.

-----

## To Test:
1. Open a milstone notifications
2. Verify that no title is shown

-----

## Regression Notes

1. Potential unintended areas of impact

    - Milestone notifications

3. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual testing

4. What automated tests I added (or what prevented me from doing so)

    - Added unit tests

-----

## PR Submission Checklist:


-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):
